src/nupic/bindings/sparse_matrix.i

```diff
@@ -940,14 +940,35 @@ def __div__(self, other):
     self->clipRowsBelowAndAbove(i.begin(), i.end(), a, b);
   }
 
-  // Returns the number of non-zeros per row, for all rows
-  PyObject* nNonZerosPerRow() const
+
+  %pythoncode %{
+    def nNonZerosPerRow(self, rows=None):
+      if rows is None:
+        return self._nNonZerosPerRow_allRows()
+      else:
+        return self._nNonZerosPerRow(numpy.asarray(rows, dtype="uint32"))
+  %}
+
+  PyObject* _nNonZerosPerRow_allRows() const
   {
     nupic::NumpyVectorT<nupic::UInt ## N1> nnzpr(self->nRows());
     self->nNonZerosPerRow(nnzpr.begin());
     return nnzpr.forPython();
   }
 
+  PyObject* _nNonZerosPerRow(PyObject* py_rows)
```

Comment:
Have you done anything to make sure there isn't a memory leak with these SWIG changes?